code/__HELPERS/reagents.dm

```diff
@@ -73,12 +73,13 @@
 		GLOB.chemical_reactions_list[primary_reagent] = list()
 	GLOB.chemical_reactions_list[primary_reagent] += R
 
-/datum/reagents/proc/create_foam(foam_volume,metaltype = 0,notification = null)
+//Creates foam from the reagent. Metaltype is for metal foam, notification is what to show people in textbox
+/datum/reagents/proc/create_foam(datum/effect_system/foamtype,foam_volume,metaltype = 0,notification = null)
 	var/location = get_turf(my_atom)
-	var/datum/effect_system/foam_spread/foam = new()
+	var/datum/effect_system/foam_spread/foam = new(foamtype)
```

Comment:
this is passing the type as argument, you want `new foamtype()` also probably want to remove the datum/effect_system type from parameter